cmd/frontend/internal/auth/accessrequest/handlers.go

```diff
@@ -0,0 +1,121 @@
+package accessrequest
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/sourcegraph/log"
+
+	"net/http"
+
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/internal/auth/userpasswd"
+	sgactor "github.com/sourcegraph/sourcegraph/internal/actor"
+	"github.com/sourcegraph/sourcegraph/internal/conf"
+	"github.com/sourcegraph/sourcegraph/internal/types"
+
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/deviceid"
+	"github.com/sourcegraph/sourcegraph/internal/errcode"
+	"github.com/sourcegraph/sourcegraph/internal/featureflag"
+	"github.com/sourcegraph/sourcegraph/internal/usagestats"
+)
+
+// HandleRequestAccess handles submission of the request access form.
+func HandleRequestAccess(logger log.Logger, db database.DB) http.HandlerFunc {
+	logger = logger.Scoped("HandleRequestAccess", "request access request handler")
+	return func(w http.ResponseWriter, r *http.Request) {
+		if handleEnabledCheck(logger, w) {
+			return
+		}
+		// Check whether builtin signup is enabled.
+		builtInAuthProvider, _ := userpasswd.GetProviderConfig()
+		if builtInAuthProvider != nil && builtInAuthProvider.AllowSignup {
+			http.Error(w, "Use sign up instead.", http.StatusConflict)
+			return
+		}
+		handleRequestAccess(logger, db, w, r)
+	}
+}
+
+// HandleRequestAccessEnabledCheck checks whether request access experimental feature is explicitly disabled
+func handleEnabledCheck(logger log.Logger, w http.ResponseWriter) (handled bool) {
```

Comment:
I'd rename this and turn the boolen around:

```suggestion
func accessRequestsFeatureEnabled(logger log.Logger, w http.ResponseWriter) (handled bool) {
```

Easier to understand. `handle...` sounds like it has side-effects and checking whether a feature is explicitly disabled is a bit weird for an experimental feature that needs to be enabled.